src/use-swr.ts

```diff
@@ -417,14 +422,10 @@ function useSWR<Data = any, Error = any>(
       config.revalidateOnMount ||
       (!config.initialData && config.revalidateOnMount === undefined)
     ) {
-      if (
-        typeof latestKeyedData !== 'undefined' &&
-        !IS_SERVER &&
-        window['requestIdleCallback']
-      ) {
+      if (typeof latestKeyedData !== 'undefined') {
         // delay revalidate if there's cache
         // to not block the rendering
-        window['requestIdleCallback'](softRevalidate)
+        rIC(softRevalidate)
```

Comment:
What if `rIC` is null?

Reply:
`useIsomorphicLayoutEffect` will execute the callback on client side only. when `IS_SERVER` is true it won't execute.

check the following tip under https://reactjs.org/docs/hooks-reference.html#uselayouteffect
> If you use server rendering, keep in mind that neither useLayoutEffect nor useEffect can run until the JavaScript is downloaded.

Reply:
My bad, I just looked at the diff, not at the context around it 🙈

Reply:
> `useIsomorphicLayoutEffect` will execute the callback on client side only. when `IS_SERVER` is true it won't execute.
> 
> check the following tip under https://reactjs.org/docs/hooks-reference.html#uselayouteffect
> 
> > If you use server rendering, keep in mind that neither useLayoutEffect nor useEffect can run until the JavaScript is downloaded.

I use SWR in WeChat mini program runtime, which `window` is undefined and `rIC` is null, but `useIsomorphicLayoutEffect` will execute. Get an error said `rIC is not a function`, search the word `rIC` lead me to this pull request.
Maybe `if(typeof latestKeyedData !== 'undefined' && rIC))` is safer than `if(typeof latestKeyedData !== 'undefined' ))`?